Fix comparison of batched_prop vs unbatched_prob in test_distributions

Flamefire opened the pull request:
When using SciPy >= 1.7 wishart_log_prob runs into singular samples which means there are `inf`s in `batched_prop` and `unbatched_prop`.
The difference of 2 `inf`s is `nan` which will fail the `equal(0` check.
However passing the tensors directly to `assertEqual` is not only supported but the correct way as it will handle `inf` values etc.

Change the same code in 2 more tests:
  - test_multivariate_normal_log_prob
  - test_lowrank_multivariate_normal_log_prob

Flamefire committed:
Fix comparison of `batched_prop` vs `unbatched_prob` in test_distributions

When using SciPy >= 1.7 wishart_log_prob runs into singular samples
which means there are `inf`s in `batched_prop` and `unbatched_prop`.
The difference of 2 `inf`s is `nan` which will fail the `equal(0` check.
However passing the tensors directly to `assertEqual` is not only
supported but the correct way as it will handle `inf` values etc.

Change the same code in 2 more tests:
  - test_multivariate_normal_log_prob
  - test_lowrank_multivariate_normal_log_prob

soulitzer reviewed (dismissed):
Thanks!

soulitzer commented:
Test failures look real - looks like you'll need to either pass both atol/rtol or not pass it in at all.

Flamefire commented:
> Test failures look real - looks like you'll need to either pass both atol/rtol or not pass it in at all.

You are right. Updated and rebased after the new CLA is now finally signed by my company.

Flamefire commented:
By accident I discovered the same code in 2 more subtests, so I fixed all of them to make it resilient against future changes

Flamefire commented:
Failing CI is unrelated:

> RuntimeError: test_jit failed! Received signal: SIGIOT

soulitzer reviewed (approved):
Sorry for the delay, looks good!

soulitzer commented:
@pytorchbot merge -f

soulitzer commented:
@pytorchbot merge -f "Unrelated failure"
